shared/product.go

```diff
@@ -0,0 +1,103 @@
+package shared
+
+import (
+	"fmt"
+
+	"github.com/rancher/distros-test-framework/config"
+)
+
+// GetProductObject returns the distro "Product" object based on the config file
+func GetProductObject() (Product, error) {
```

Comment:
Why do u need a get product if we already have in aux.go?

Reply:
this one returns the Product object with which you can perform operations product.CertRotate etc. the other returns a string.

Reply:
ok , so we should use one or another 

i will add some important points here also for this file:

- if this is kind of a implementation of something ( which is , this is not some aux or helper  functions ) and some methods ( for product ) it should live on /PKG , not here. on pkg should live our internal or custom implementations


- i would not tie all this up to a type string ( product ) this adds no value + you are  coupling more your implementation with this type needing now to pass around all functions as methods (  func ()  name () {}  ) 
and also whenever we are going to use the implementation we would need to import this type as dependencie


- GetSystemCtlCmd and ManageService also on the same subject , in a design perpespecive do not make much sense live in a product implementation , they dont even need.  a product to work , even though needed a product they are generic enough to not make part on the product implementation 

- i really like the ideia of adding a product implementation itself but in here we are adding another function that do the same of the existing one ( getproduct) just return a type instead of string , also we are adding functions not related


summary :

- if u are going to create a implementation should live under /pkg
- u should add those there and refactor and improve if u want 

```

// GetProduct returns the distro product based on the config file
func GetProduct() (string, error) {
	cfgPath, err := EnvDir(".")
	if err != nil {
		return "", ReturnLogError("failed to get config path: %v\n", err)
	}

	cfg, err := config.AddConfigEnv(cfgPath)
	if err != nil {
		return "", ReturnLogError("failed to get config: %v\n", err)
	}
	if cfg.Product != "k3s" && cfg.Product != "rke2" {
		return "", ReturnLogError("unknown product")
	}

	return cfg.Product, nil
}

// GetProductVersion return the version for a specific distro product
func GetProductVersion(product string) (string, error) {
	if product != "rke2" && product != "k3s" {
		return "", ReturnLogError("unsupported product: %s\n", product)
	}
	version, err := getVersion(product + " -v")
	if err != nil {
		return "", ReturnLogError("failed to get version for product: %s, error: %v\n", product, err)
	}

	return version, nil
}

```

OR

keep under aux funcs as you were doing before 

in both cases i would suggest not tie this up with a simple type as string, u are going to create dependencies not needed or not worthy.

Reply:
per our conversation, moved the GetProduct and GetProductVersion methods to product.go as well
changed all of these to static methods for now. - passing the product var as a string. 
if we want to make this an implementation in pkg, we will address this in the future in another PR and keep this short and simple.